packages/account-v2/src/containers/PersonalDetailsForm/validation.ts

```diff
@@ -0,0 +1,151 @@
+
+type initialValues = {
+    addressCity?: string;
+    addressLine1?: string;
+    addressLine2?: string;
+    addressPostcode?: string;
+    addressState?: string;
+    dateOfBirth?: number | null;
+    email?: string;
+    emailConsent?: 0 | 1;
+    employmentStatus?: string;
+    firstName?: string;
+    lastName?: string;
+    phone?: string | null;
+    placeOfBirth?: string;
+    requestProfessionalStatus?: 1;
+    residence?: string | null;
+    taxIdentificationNumber?: string;
+};
+
+const min2max50text = 'You should enter 2-50 characters.';
+const lessThan70Text = 'Should be less than 70.';
+
+const getBaseSchema = () =>
+    Yup.object().shape({
+        addressCity: Yup.string()
+            .required('Town/City is required.')
+            .max(70, lessThan70Text)
+            .matches(
+                /^[A-Za-z]+(?:[.' -]*[A-Za-z]+){1,70}$/,
+                'Only letters, space, hyphen, period, and apostrophe are allowed.'
+            ),
+        addressLine1: Yup.string()
+            .trim()
+            .required('First line of address is required.')
+            .max(70, lessThan70Text)
+            .matches(/^[\p{L}\p{Nd}\s'.,:;()\u00b0@#/-]{0,70}$/u, 'Use only the following special characters: 70'),
+        addressLine2: Yup.string()
+            .trim()
+            .max(70, lessThan70Text)
+            .matches(/^[\p{L}\p{Nd}\s'.,:;()\u00b0@#/-]{0,70}$/u, 'Use only the following special characters: 70'),
+        addressPostcode: Yup.string()
+            .max(20, 'Please enter a Postal/ZIP code under 20 chatacters.')
+            .matches(/^[A-Za-z0-9][A-Za-z0-9\s-]*$/, 'Only letters, numbers, space, and hyphen are allowed.'),
+        firstName: Yup.string()
+            .required('First name is required.')
+            .min(2, min2max50text)
+            .max(50, min2max50text)
+            .matches(/^(?!.*\s{2,})[\p{L}\s'.-]{2,50}$/u, 'Letters, spaces, periods, hyphens, apostrophes only.'),
+        lastName: Yup.string()
+            .required('Last name is required.')
+            .min(2, min2max50text)
+            .max(50, min2max50text)
+            .matches(/^(?!.*\s{2,})[\p{L}\s'.-]{2,50}$/u, 'Letters, spaces, periods, hyphens, apostrophes only.'),
+        phone: Yup.string()
+            .required('Phone is required.')
+            .min(9, 'You should enter 9-35 numbers.')
+            .max(35, 'You should enter 9-35 characters.')
+            .matches(/^\+?([0-9-]+\s)*[0-9-]+$/, 'Please enter a valid phone number (e.g. +15417541234).'),
+    });
+
+const getLocation = (location_list: StatesList, value: string, type: keyof StatesList[number]) => {
+    if (!value || !location_list.length) return '';
+    const locationObj = location_list.find(
+        location => location[type === 'text' ? 'value' : 'text']?.toLowerCase() === value.toLowerCase()
+    );
+
+    return locationObj?.[type] ?? '';
+};
+
+export const getPersonalDetailsInitialValues = (
```

Comment:
Lets try to get default values from Yup schema itself. U can check `getNameDOBValidationSchema` and in Formik, u can use `schema.getDefault()`. Will reduce a lot of initialValue objects

Reply:
I think we shall remain initialValue function for PersonalDetails because there's a lot of logics handling inside the `getPersonalDetailsValue` if we use `Yup.cast()` we need to over complicate the function which is not really a good approach.